src/main/java/com/bookstore/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,113 @@
+package com.bookstore.service;
+
+import com.bookstore.dto.ShoppingCartDto;
+import com.bookstore.dto.UpdateCartItemDto;
+import com.bookstore.entity.Book;
+import com.bookstore.entity.CartItem;
+import com.bookstore.entity.ShoppingCart;
+import com.bookstore.entity.User;
+import com.bookstore.exception.EntityNotFoundException;
+import com.bookstore.mapper.ShoppingCartMapper;
+import com.bookstore.repository.BookRepository;
+import com.bookstore.repository.CartItemRepository;
+import com.bookstore.repository.ShoppingCartRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+
+    @Override
+    public void createShoppingCart(User user) {
+        ShoppingCart shoppingCart = new ShoppingCart();
+        shoppingCart.setUser(user);
+        shoppingCartRepository.save(shoppingCart);
+    }
+
+    @Override
+    public ShoppingCartDto addBookToShoppingCart(UpdateCartItemDto itemDto, User user) {
+        if (itemDto.getQuantity() <= 0) {
```

Comment:
You have `@Positive` validation for UpdateCartItemDto quantity, so this checking is redundant